python/nano/src/bigdl/nano/pytorch/inference/optimizer.py

```diff
@@ -750,17 +784,35 @@ def _throughput_calculate_helper(iterrun, baseline_time, func, *args):
     return np.mean(time_list) * 1000, True
 
 
+def _signature_check(function):
+    '''
+    A quick helper to judge whether input function is following this calling
+    method `metric(pred, target)`.
+    '''
+    import inspect
+    sig = inspect.signature(function)
+    if len(sig.parameters.values()) != 2:
+        return False
+    param1_name = list(sig.parameters.values())[0].name
+    param2_name = list(sig.parameters.values())[0].name
+    if "pred" in param1_name:
+        return True
+    if "target" in param2_name:
+        return True
+    return False
+
+
 def _accuracy_calculate_helper(model, metric, data):
     '''
     A quick helper to calculate accuracy
     '''
-    metric_list = []
-    sample_num = 0
-    with torch.no_grad():
-        for i, (data_input, target) in enumerate(data):
-            metric_list.append(metric(model(data_input), target).numpy() * data_input.shape[0])
-            sample_num += data_input.shape[0]
-    return np.sum(metric_list) / sample_num
+    if isinstance(metric, Metric) or _signature_check(metric) is True:
```

Comment:
What if data is None?

Reply:
will add ` invalidInputError(data is not None,xxx)` for this case.
Error message will look like this:
![image](https://user-images.githubusercontent.com/105281011/193222904-9fb5b12e-59dd-462f-8ce3-215690a14d3d.png)
Do you think it's ok ?

Reply:
ok